linter/parsetemplates/templates_test.go

```diff
@@ -157,6 +157,14 @@ func TestListTemplatePathsFromTemplates(t *testing.T) {
 				"test2",
 			},
 		},
+		{
```

Comment:
These new test cases validated that we now detect the variable usage in more cases.